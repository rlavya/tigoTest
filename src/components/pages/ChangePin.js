import React, { Component } from "react";
import RaisedButton from "material-ui/RaisedButton";
import TextField from "material-ui/TextField";
import PageTitle from "../components/PageTitle";
import { lightBlue200, indigo300 } from "material-ui/styles/colors";
import get from "lodash/get";
import { connect } from "react-redux";
import {
  updateNewPinAction,
  updateOldPinAction,
  changePinAction
} from "../../actionCreators/auth.js";
import { onEnterAction } from "../../util";
import styled from "styled-components";

const InputWrap = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  img {
    display: block;
    margin-bottom: -23px;
    margin-right: 10px;
    margin-left: -30px;
  }
`;

export class ChangePin extends Component {
  render() {
    const changePinIfValid = onEnterAction(() => {
      if (this.props.oldPin.trim() !== "" && this.props.newPin.trim() !== "") {
        this.props.changePin();
      }
    });

    return (
      <PageTitle page="ChangePin">
        <div className="max-width-4 mx-auto">
          <div className="clearfix center">
            <h1 className="light text-primary h2">Ingresa tus credenciales:</h1>

            <div className="masked-input">
              <InputWrap>
                <img src="/images/icn-password.png" alt="" />
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN ACTUAL"
                  type="tel"
                  pattern="[0-9]*"
                  id="oldPin"
                  value={this.props.oldPin}
                  onChange={this.props.updateOldPin}
                  onKeyDown={changePinIfValid}
                />
              </InputWrap>
            </div>
            <br />
            <div className="masked-input">
              <InputWrap>
                <img src="/images/icn-password-new.png" alt="" />
                <TextField
                  underlineFocusStyle={{ borderColor: lightBlue200 }}
                  floatingLabelStyle={{ color: indigo300 }}
                  floatingLabelText="PIN NUEVO"
                  type="tel"
                  pattern="[0-9]*"
                  id="newPin"
                  value={this.props.newPin}
                  onChange={this.props.updateNewPin}
                  onKeyDown={changePinIfValid}
                />
              </InputWrap>
            </div>
            <br />

            <RaisedButton
              className="mt2"
              secondary={true}
              label="Cambiar"
              onClick={this.props.changePin}
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={
                this.props.oldPin.trim() === "" ||
                this.props.newPin.trim() === ""
              }
            />
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      oldPin: get(state, "auth.pinChange.old") || "",
      newPin: get(state, "auth.pinChange.new") || ""
    };
  },
  dispatch => {
    return {
      updateOldPin: (e, value) => {
        return dispatch(updateOldPinAction(value));
      },
      updateNewPin: (e, value) => {
        return dispatch(updateNewPinAction(value));
      },
      changePin: (e, value) => {
        return dispatch(changePinAction());
      }
    };
  }
)(ChangePin);
