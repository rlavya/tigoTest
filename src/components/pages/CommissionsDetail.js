import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import { redirectTo } from "../../router";
import TransactionsOrCommissionsDetail from "../components/TransactionsOrCommissionsDetail";

export default connect(
  state => {
    const items = dataForId(
      state,
      configToId("CommissionDetail", "list", true)
    );
    return {
      items,
      pageName: siteTree.TransactionsDetail.name,
      periodFilter: state.route.periodFilter || "hora",
      transactionFilter: state.route.transactionFilter || "todas"
    };
  },
  dispatch => {
    return {
      changePeriodFilter(filter) {
        redirectTo({ periodFilter: filter }, true);
      },
      changeTransactionFilter(filter) {
        redirectTo({ transactionFilter: filter }, true);
      }
    };
  }
)(TransactionsOrCommissionsDetail);
