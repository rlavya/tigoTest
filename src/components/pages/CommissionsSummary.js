import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import TransactionsOrCommissionsSummary from "../components/TransactionsOrCommissionsSummary";

export default connect(state => {
  const commissions = dataForId(state, configToId("Commission", "list", true));

  return {
    items: commissions,
    pageName: siteTree.CommissionsSummary.name,
    detailViewPageName: siteTree.CommissionsDetail.name
  };
})(TransactionsOrCommissionsSummary);
