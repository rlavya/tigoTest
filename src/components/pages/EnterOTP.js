import React, { Component } from "react";
import PageTitle from "../components/PageTitle";
import { lightBlue200 } from "material-ui/styles/colors";
import RaisedButton from "material-ui/RaisedButton";
import get from "lodash/get";
import { connect } from "react-redux";
import TextField from "material-ui/TextField";
import styled from "styled-components";

import {
  requestOtpAction,
  updateOTPCodeAction,
  validateOTPCodeAction
} from "../../actionCreators/auth.js";

import { onEnterAction } from "../../util";

const TextFieldWrap = styled.div`
  margin: 0 auto;
  width: 157px;
`;

export class EnterOTP extends Component {
  render() {
    const onEnter = onEnterAction(() => {
      this.props.validateCode();
    });
    return (
      <PageTitle page="OTP">
        <div className="max-width-4 mx-auto">
          <div className="center">
            <h1 className="light text-primary h2">
              Ingresa el código <br />de verificación
            </h1>
            <div className="mt3">
              <p className="regular text-primary inline">
                Te enviamos un SMS al{" "}
              </p>
              <p className="bold text-primary inline">+{this.props.msisdn}</p>
            </div>
            <TextFieldWrap>
              <TextField
                value={this.props.otpCode}
                onChange={this.props.updateCode}
                id="validate-otp"
                className="mt4 mb1"
                type="tel"
                pattern="[0-9]*"
                fullWidth={true}
                onKeyDown={onEnter}
                underlineStyle={{
                  backgroundImage: "url(/images/segmented_inactive.png)",
                  backgroundRepeat: "no-repeat",
                  backgroundPosition: "2px 0px",
                  height: "2px",
                  border: "none",
                  bottom: "1px"
                }}
                underlineFocusStyle={{
                  backgroundImage: "url(/images/segmented_active.png)",
                  backgroundRepeat: "no-repeat",
                  backgroundPosition: "2px 0px",
                  height: "2px",
                  border: "none",
                  bottom: "1px"
                }}
                inputStyle={{
                  fontSize: "2.5rem",
                  fontFamily: "'Roboto Mono', monospace",
                  letterSpacing: "2px"
                }}
              />
            </TextFieldWrap>
            <br />
            <RaisedButton
              label="Continuar"
              secondary={true}
              className="mt2 mb3"
              disabledBackgroundColor="#CAF1F6"
              disabledLabelColor="#FFF"
              disabled={this.props.otpCode.length < 6}
              onClick={this.props.validateCode}
            />
            <br />
            <p className="inline text-primary mt2">¿No recibiste el código?</p>
            <p
              className="inline pointer"
              style={{ color: lightBlue200 }}
              onClick={this.props.requestNewCode}
            >
              {" "}
              Enviar de nuevo.
            </p>
          </div>
        </div>
      </PageTitle>
    );
  }
}

export default connect(
  state => {
    return {
      msisdn: get(state, "auth.otp.msisdn") || "",
      otpCode: get(state, "auth.otpCode") || ""
    };
  },
  dispatch => {
    return {
      updateCode: (e, value) => {
        return dispatch(updateOTPCodeAction((value || "").trim()));
      },
      validateCode: () => {
        return dispatch(validateOTPCodeAction());
      },
      requestNewCode: () => {
        return dispatch(requestOtpAction({}));
      }
    };
  }
)(EnterOTP);
