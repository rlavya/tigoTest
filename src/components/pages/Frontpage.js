import React, { Component } from "react";
import { List, ListItem } from "material-ui/List";
import ChevronRight from "material-ui/svg-icons/navigation/chevron-right";
import Divider from "material-ui/Divider";
import { redirectTo } from "../../router";

export class Frontpage extends Component {
  render() {
    return (
      <div className="max-width-4 mx-auto bg-primary height100 flex flex-column">
        <img
          src="http://via.placeholder.com/250x250"
          width="250"
          height="250"
          className="mx-auto my2"
          alt=""
        />
        <List>
          <Divider style={{ backgroundColor: "#406A9C" }} />
          <ListItem
            primaryText={
              <div>
                <b>PDV</b> - Punto de Venta
              </div>
            }
            rightIcon={<ChevronRight color="#4CC5F2" />}
            onClick={() => {
              redirectTo({ page: "Login" });
            }}
            style={{ color: "#fafafa" }}
          />
          <Divider style={{ backgroundColor: "#406A9C" }} />
          <ListItem
            primaryText={
              <div>
                <b>PDA</b> - Punto de Activaci√≥n
              </div>
            }
            rightIcon={<ChevronRight color="#4CC5F2" />}
            onClick={() => {
              redirectTo({ page: "Login" });
            }}
            style={{ color: "#fafafa" }}
          />
          <Divider style={{ backgroundColor: "#406A9C" }} />
        </List>
      </div>
    );
  }
}

export default Frontpage;
