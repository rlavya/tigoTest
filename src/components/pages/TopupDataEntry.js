import React, { Component } from "react";
import PageTitle from "../components/PageTitle";
import FlatButton from "material-ui/FlatButton";
import { redirectTo } from "../../router";
import { lightBlue200 } from "material-ui/styles/colors";
import filter from "lodash/filter";
import map from "lodash/map";
import { connect } from "react-redux";
import { siteTree } from "../../siteTree";
import { dataForId, configToId } from "../../util/db";
import Checkbox from "material-ui/Checkbox";
import { Card, CardTitle, CardText } from "material-ui/Card";
import RadioButtonCheckedIcon from "material-ui/svg-icons/toggle/radio-button-checked";
import RadioButtonUncheckedIcon from "material-ui/svg-icons/toggle/radio-button-unchecked";
import styled from "styled-components";
import get from "lodash/get";
import { formatPhoneNumber } from "../../util/index";

const CardWrap = styled.div`
  padding: 15px;
`;

const ItemWrap = styled.div`
  width: 50%;
  float: left;
  font-size: 150%;
  padding: 5px 0;
`;

const Items = ({ items, selected }) => {
  return (
    <div className="clearfix">
      {map(items, i => {
        return (
          <ItemWrap key={i.id}>
            <Checkbox
              label={"Q" + i.price}
              checked={i.id === selected}
              checkedIcon={
                <RadioButtonCheckedIcon
                  style={{ color: lightBlue200, fill: lightBlue200 }}
                />
              }
              uncheckedIcon={
                <RadioButtonUncheckedIcon
                  style={{ color: lightBlue200, fill: lightBlue200 }}
                />
              }
              onCheck={() => {
                redirectTo({ id: i.id }, true);
              }}
            />
          </ItemWrap>
        );
      })}
    </div>
  );
};

const Loading = props => {
  return (
    <CardWrap>
      <Card>
        <CardTitle
          style={{ background: "#2e60aa", color: "white" }}
          subtitleColor="white"
          subtitle={"..."}
        />
        <CardText>
          <div className="clearfix">
            {map(new Array(3), (_, i) => {
              return (
                <ItemWrap key={i}>
                  <Checkbox
                    label={
                      <span
                        style={{
                          width: "60px",
                          height: "10px",
                          background: "#ccc",
                          display: "inline-block",
                          borderRadius: "5px"
                        }}
                      />
                    }
                    checkedIcon={
                      <RadioButtonCheckedIcon
                        style={{ color: lightBlue200, fill: lightBlue200 }}
                      />
                    }
                    uncheckedIcon={
                      <RadioButtonUncheckedIcon
                        style={{ color: lightBlue200, fill: lightBlue200 }}
                      />
                    }
                  />
                </ItemWrap>
              );
            })}
          </div>
        </CardText>
      </Card>
    </CardWrap>
  );
};

const RenderItemGroup = ({ title, items, selected }) => {
  if (!items || !items.length) {
    return null;
  }
  return (
    <CardWrap>
      <Card>
        <CardTitle
          style={{ background: "#2e60aa", color: "white" }}
          subtitleColor="white"
          subtitle={title}
        />
        <CardText>
          <Items items={items} selected={selected} />
        </CardText>
      </Card>
    </CardWrap>
  );
};

export class TopupDataEntry extends Component {
  render() {
    return (
      <PageTitle page={this.props.pageName}>
        <div className="max-width-4 mx-auto">
          {this.props.items.state === "loading" ? (
            <Loading />
          ) : (
            <div>
              <RenderItemGroup
                title={"RECOMENDADOS PARA " + this.props.clientPhoneNumber}
                items={this.props.suggestedItems}
                selected={this.props.selected}
              />
              <RenderItemGroup
                title={"TODAS LAS RECARGAS"}
                items={this.props.notSuggestedItems}
                selected={this.props.selected}
              />
              {this.props.hasItems ? (
                <div className="center">
                  <FlatButton
                    secondary={true}
                    label="Siguiente"
                    disabled={!this.props.topupId}
                    onClick={() => {
                      redirectTo({
                        page: "TopupReview",
                        id: this.props.selected
                      });
                    }}
                  />
                </div>
              ) : (
                ""
              )}
            </div>
          )}
        </div>
      </PageTitle>
    );
  }
}

export default connect(state => {
  const topups = dataForId(state, configToId("Topup", "list", true));
  const topupId = get(state, "route.id");

  const hasItems = (get(topups, "data") || []).length > 0;

  const suggestedItems = filter(topups.data, t => {
    return t.suggested === "true";
  });

  const notSuggestedItems = filter(topups.data, t => {
    return t.suggested === "false";
  });

  const clientPhoneNumber = formatPhoneNumber(
    get(state, "clientPhone.phoneNumber")
  );

  return {
    items: topups,
    suggestedItems,
    notSuggestedItems,
    pageName: siteTree.TopupDataEntry.name,
    selected: state.route.id,
    clientPhoneNumber,
    topupId,
    hasItems
  };
})(TopupDataEntry);
