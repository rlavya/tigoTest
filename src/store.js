import { applyMiddleware, createStore } from "redux";
import { createLogger } from "redux-logger";
import { composeWithDevTools } from "redux-devtools-extension/developmentOnly";
import thunk from "redux-thunk";
import reducer from "./reducer";

const getMiddleware = () => {
  return applyMiddleware(createLogger(), thunk);
};

const store = createStore(reducer, composeWithDevTools(getMiddleware()));

export default store;
