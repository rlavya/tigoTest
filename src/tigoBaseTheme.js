const tigoBaseTheme = {
  fontFamily: "Roboto, sans-serif",
  borderRadius: 2,
  palette: {
    primary1Color: "#00377B",
    primary2Color: "#4CC5F2",
    primary3Color: "#333",
    accent1Color: "#4CC5F2",
    accent2Color: "#4CC5F2",
    accent3Color: "#4CC5F2",
    textColor: "#00377B",
    secondaryTextColor: "#839ebf",
    alternateTextColor: "#fafafa",
    canvasColor: "#FFFFFF"
  }
};

export default tigoBaseTheme;
