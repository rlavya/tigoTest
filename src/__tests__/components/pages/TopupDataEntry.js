import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import getMuiTheme from "material-ui/styles/getMuiTheme";
import tigoBaseTheme from "../../../tigoBaseTheme";
import { TopupDataEntry } from "../../../components/pages/TopupDataEntry";
import renderer from "react-test-renderer";
import React from "react";
import store from "../../../store";
import { Provider } from "react-redux";

it("renders correctly", () => {
  const rendered = renderer.create(
    <MuiThemeProvider muiTheme={getMuiTheme(tigoBaseTheme)}>
      <Provider store={store} >
        <TopupDataEntry items={{state: "loaded"}}/>
      </Provider>
    </MuiThemeProvider>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
