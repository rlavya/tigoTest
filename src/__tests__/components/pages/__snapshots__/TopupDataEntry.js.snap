// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="max-width-4 mx-auto"
>
  <div>
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#00377B",
          "borderRadius": "0px",
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "#00377B",
          "display": "flex",
          "fontFamily": "Roboto, sans-serif",
          "muiPrepared": true,
          "paddingLeft": 24,
          "paddingRight": 24,
          "position": "relative",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
          "zIndex": 1100,
        }
      }
    >
      <div
        style={
          Object {
            "marginLeft": -16,
            "marginRight": 8,
            "marginTop": 8,
            "muiPrepared": true,
          }
        }
      >
        <span />
      </div>
      <h1
        onClick={[Function]}
        style={
          Object {
            "boxFlex": 1,
            "color": "#fafafa",
            "flex": "1",
            "fontSize": 24,
            "fontWeight": 400,
            "height": 64,
            "letterSpacing": 0,
            "lineHeight": "64px",
            "margin": 0,
            "muiPrepared": true,
            "overflow": "hidden",
            "paddingTop": 0,
            "textOverflow": "ellipsis",
            "whiteSpace": "nowrap",
          }
        }
      >
        
      </h1>
    </div>
  </div>
  <div>
    <div
      className="center"
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "#4CC5F2",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <span
            style={
              Object {
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "muiPrepared": true,
                "paddingLeft": 16,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "verticalAlign": "middle",
              }
            }
          >
            Siguiente
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
